modules/core/resources/META-INF/smithy/jsonunknown.smithy

```diff
@@ -0,0 +1,10 @@
+$version: "2"
+
+namespace alloy
+
+/// Retain unknown fields of a containing structure in this document member
+@trait(
+    selector: "structure > member :test(> map > member > document)"
```

Comment:
So, to answer your question, I'd like for this to be eventually extended to allow for annotating union members, to allow for capturing unknown values.  

As long as we can amend the selector to also cater to that use-case, I'm fine with it.

Reply:
Once we want to extend this trait to support unions, we can update the selector to 
```:is(structure > member :test(> map > member > document), union > member :test(> document))```
This will allow the trait to be applied to a union member targeting a `Document`.